qualtran/_infra/bloq.py

```diff
@@ -166,7 +167,7 @@ def as_composite_bloq(self) -> 'CompositeBloq':
         bb, initial_soqs = BloqBuilder.from_signature(self.signature, add_registers_allowed=False)
         return bb.finalize(**bb.add_d(self, **initial_soqs))
 
-    def adjoint(self) -> 'Bloq':
+    def adjoint(self) -> 'Adjoint':
```

Comment:
Is this correct? `bloq.adjoint()` in general need not return an `Adjoint` right?

Reply:
How about GateWithRegisters?  There are quite a bit of places where there is `bloq.adjoint() ** 2` which requires a GateWithRegisters.  (Bloq has no pow)

Reply:
Update: reverted.  Found a different way to fix it.